File: src/Plotly.NET/Traces/ObjectAbstractions/Meanline.fs
namespace Plotly.NET.TraceObjects

open Plotly.NET
open Plotly.NET.LayoutObjects
open DynamicObj
open System
open System.Runtime.InteropServices

/// Meanline type inherits from dynamic object (parent violin)
type Meanline () =
    inherit DynamicObj ()

    /// Initialized Line object
    static member init
        (
            [<Optional;DefaultParameterValue(null)>] ?Visible: bool,
            [<Optional;DefaultParameterValue(null)>] ?Color: Color,
            [<Optional;DefaultParameterValue(null)>] ?Width: float
        ) =
            Meanline () 
            |> Meanline.style
                (
                    ?Visible    = Visible,
                    ?Color      = Color  ,
                    ?Width      = Width             
                )


    // Applies the styles to Line()
    static member style
        (
            [<Optional;DefaultParameterValue(null)>] ?Visible: bool,
            [<Optional;DefaultParameterValue(null)>] ?Color: Color,
            [<Optional;DefaultParameterValue(null)>] ?Width: float
        ) =
            (fun (line:Meanline) -> 
                ++? ("visible", Visible    )
                ++? ("color", Color      )
                ++? ("width", Width      )
                    
                // out -> 
                line
            )



               

File: src/Plotly.NET.CSharp/ObjectAbstractions/TraceObjects/MeanLine.cs

using Microsoft.FSharp.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Plotly.NET.CSharp;

public static partial class TraceObjects
{
    public static Plotly.NET.TraceObjects.MeanLine InitMeanLine
        (
            Optional<bool> Visible = default,
            Optional<Color> Color = default,
            Optional<double> Width = default
        ) 
            =>
                Plotly.NET.TraceObjects.MeanLine.init(
                    Visible: Visible.ToOption(),
                    Color: Color.ToOption(),
                    Width: Width.ToOption()
                );
}

